pkg/remotecmd/interface.go

```diff
@@ -0,0 +1,51 @@
+package remotecmd
+
+import (
+	devfilev1 "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+
+	"github.com/redhat-developer/odo/pkg/kclient"
+)
+
+// RemoteProcessStatus is an enum type for representing process statuses.
+type RemoteProcessStatus int
+
+const (
+	// Unknown represents a process for which the status cannot be determined reliably or is not handled yet by us.
+	Unknown RemoteProcessStatus = iota + 1
+
+	// Stopped represents a process stopped.
+	Stopped
+
+	// Running represents a running process.
+	Running
+)
+
+// RemoteProcessHandler is an interface for managing processes that are intended to be executed remotely,
+// in Kubernetes/OpenShift containers, depending on the selected strategy (with or without Supervisord for the moment).
+// See GetRemoteProcessHandlerInstance for retrieving concrete instances of RemoteProcessHandler
+type RemoteProcessHandler interface {
+
```

Comment:
I got this answered while reading the code, but I would expect it to be a part of this interface because it's providing methods to a bunch of things for a process except getting the process info itself.